src/main/java/de/uni_hamburg/informatik/swt/se2/kino/wertobjekte/Geldbetrag.java

```diff
@@ -0,0 +1,267 @@
+     *     "-123131,2"  (123131 Euro und 20 Cent, negativ)
+     *     "123,"       (123 Euro und 0 Cent)
+     *  3. Leerer String, es wird {@link Geldbetrag.ZERO } zurueckgegeben
+     *
+     * Maximalwert ist hierbei Geldbetrag.MAX_VALUE und Minimalwert Geldbetrag.MIN_VALUE
+     * Falls der eingegebene String diese Grenzen überschreitet, wird dementsprechend der jeweilige Extremwert zurückgegeben
+     * Beispiele:
+     * "1367736176312671678,13" => Geldbetrag.MAX_VALUE
+     * "-3113221123131312" => Geldbetrag.MIN_VALUE
+     * 
+     * @param string den string zum uebersetzen
+     * @return den Geldbetrag der von string uebersetzt wurde, {@link Optional.empty()} falls der String in einem ungueltigen Format ist.
+     * 
+     * @require string != null
+     * @require string.split(",").length <= 1
+     */
+    public static Optional<Geldbetrag> fromString(String string) {
+        assert string != null : "Vorbedingung verletzt: string != null";
+        
+        String[] parts = string.split(",", 2);
+        
+        String firstHalf = parts.length > 0 ? parts[0] : "";
+        if (!firstHalf.matches("-{0,1}\\d*"))
+        {
+            return Optional.empty();
+        }
+        
+        String secondHalf = parts.length > 1 ? parts[1] : "";
+        if (!secondHalf.matches("\\d{0,2}"))
+        {
+            return Optional.empty();
+        }
+        
+        // '-' am Anfang muss sowohl für Euro als auch für Cent gelten!
+        // Sonst wird -300,21 als -30000 + 21 Eurocent interpretiert!
+        // Einfach am Ende mit -1 multiplizieren, falls '-' am Anfang!
+        boolean negativ = firstHalf.startsWith("-");
+        
+        // entfernt das '-' als präfix
+        if (negativ)
+        {
+            firstHalf = firstHalf.substring(1);
+        }
+        
+        int euro = 0;
+        if (!firstHalf.isEmpty())
+        {
+            try
+            {
+                euro = Integer.parseInt(firstHalf);
+            }
+            catch (NumberFormatException e)
+            {
+                return Optional.empty();
+            }
+        }
+        
+        // eine Ziffer als Nachkommastelle
+        // z.B. .....,9 sollte als .... Euro + 90 Cent interpretiert werden!
+        // Ansatz: 0 anhängen, also aus "9" wird "90"
+        // Mithilfe von StringUtils.pad mit padDirection = LeftRight.Right rechts 0'en ranfügen, bis man 2 Stellen hat!
+        // Dann wird auch "" zu "00" gepadded und kann als int geparsed werden!
+        String paddedSecondHalf = StringUtils.pad(secondHalf, 2, '0', LeftRight.RIGHT);
+        
+        // hier sollte niemals eine NumberFormatException auftreten
+        // da vorher ja per regex geprüft wird, dass nur Ziffern drin sind
+        // und der String mit 0en auf 2 Stellen gefüllt wird!
+        int cent = Integer.parseInt(paddedSecondHalf);
+        
+        if (cent >= 100)
+        {
+            return Optional.empty();
+        }
+        
```

Comment:
ab hier könnten wir ja eigendlich auch cent und euro in zwei geldbeträge konvertieren und mit denen die operationen machen. dann musst du das ganze nicht auch hier machen.